### PR TITLE
Use 3.1.1 versions in blazor-wasm

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Review by dougbu — APPROVED** [review]
If this passes validation builds, great. Otherwise, please close this PR and add your commit to #18280. That baseline update will do the needful to ensure this works.

**Comment by pranavkm** [comment]
Thanks @dougbu. We're using this as a fallback in case the other PR needs further time to stabilize.

**Review by javiercn — APPROVED** [review]
Provided build passes

**Comment by wtgodbe** [comment]
I probably won't get my PR instead, so if this works as expected, let's take this for preview1. Please crack open some binaries & nupkgs from this to make sure that they're referencing 3.1.1 packages & 3.1.0.0 AssemblyVersions

**Comment by pranavkm** [comment]
> [ERROR] A fatal error was encountered. The library 'hostpolicy.dll' required to execute the application was not found in 'F:\workspace\_work\1\s\.dotnet\shared\Microsoft.AspNetCore.App\3.1.1'.

Might need a little more work to get going.

**Comment by wtgodbe** [comment]
I tried updating to the 3.1.1 SDK

**Comment by wtgodbe** [comment]
What's up with the E2E tests? The build says they failed, but AzDO lists them under "other" with no logs. E.g. `Render_HtmlBlock_Integration`

**Comment by wtgodbe** [comment]
At a glance, the packages/assemblies look good. `Microsoft.AspNetCore.Blazor` depends on the 3.1.1 versions of `Microsoft.AspNetCore.Components.Web` and `Microsoft.Extensions.Configuration`, and the 3.2.0-preview1 version of `Mono.WebAssembly.Interop`. However, the assembly in the package appears to be referencing ref assemblies from AspNetCore, but implementation assemblies from Extensions (e.g. Microsoft.Extensions.DependencyInjection.Abstractions version 3.1.1.0) - There's likely some work we did in 3.1 that didn't make it into this branch (CC @dougbu, @JunTaoLuo, @Pilchie)

File: src/ProjectTemplates/test/BlazorWasmTemplateTest.cs [hunk 1]
```diff
@@ -39,19 +39,19 @@ public async Task BlazorWasmStandaloneTemplate_Works()
             var buildResult = await project.RunDotNetBuildAsync();
             Assert.True(0 == buildResult.ExitCode, ErrorMessages.GetFailedProcessMessage("build", project, buildResult));
 
-            await BuildAndRunTest(project.ProjectName, project);
+            // await BuildAndRunTest(project.ProjectName, project);
```
Comment (ryanbrandenburg) [inline src/ProjectTemplates/test/BlazorWasmTemplateTest.cs]: This is effectively the same thing as disabling the test, all of the interesting things happen in BuildAndRunTests and TestBasicNavigation. If we MUST do this for some reason at least file an issue about re-instating it.

**Review by ryanbrandenburg — CHANGES_REQUESTED** [review]

**Comment by JunTaoLuo** [comment]
@wtgodbe maybe the changes are stuck in https://github.com/dotnet/aspnetcore/pull/17873

**Comment by pranavkm** [comment]
I spent a couple of iterations trying to resolve this, but it seems fairly non-trivial. In particular edges where we reference implementation assemblies (transitively via Blazor's reference to AspNetCore) and reference assemblies result in a gnarly mess of version conflicts. I wonder if we should just remove project references from the equation in the blazor branch and instead reference packages for things that are already shipped. As it is, version updates are manual in the 3.1 branch. It might be a less complicated way to go about it.

**Pull request closed without merging**